mlflow/sklearn/utils.py

```diff
@@ -691,41 +677,17 @@ def first_custom_rank_column(df):
         # metrics for each training split, which is fairly verbose; accordingly, we filter
         # out per-split metrics in favor of aggregate metrics (mean, std, etc.)
         excluded_metric_prefixes = ["param", "split"]
-        metric_batches_to_log = _chunk_dict(
-            {
-                key: value
-                for key, value in result_row.iteritems()
-                if not any([key.startswith(prefix) for prefix in excluded_metric_prefixes])
-                and isinstance(value, Number)
-            },
-            chunk_size=min(
-                MAX_ENTITIES_PER_BATCH - MAX_PARAMS_TAGS_PER_BATCH, MAX_METRICS_PER_BATCH
-            ),
+        metrics_to_log = {
+            key: value
+            for key, value in result_row.iteritems()
+            if not any([key.startswith(prefix) for prefix in excluded_metric_prefixes])
+            and isinstance(value, Number)
+        }
+        autologging_client.log_metrics(
+            run_id=pending_child_run_id, metrics=metrics_to_log,
         )
 
-        for params_batch, metrics_batch in zip_longest(
-            param_batches_to_log, metric_batches_to_log, fillvalue={}
-        ):
-            # Trim any parameter keys / values and metric keys that exceed the limits
-            # imposed by corresponding MLflow Tracking APIs (e.g., LogParam, LogMetric)
-            truncated_params_batch = _truncate_dict(
-                params_batch, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH
-            )
-            truncated_metrics_batch = _truncate_dict(
-                metrics_batch, max_key_length=MAX_ENTITY_KEY_LENGTH
-            )
-            client.log_batch(
-                run_id=child_run.info.run_id,
-                params=[
-                    Param(str(key), str(value)) for key, value in truncated_params_batch.items()
-                ],
-                metrics=[
-                    Metric(key=str(key), value=value, timestamp=child_run_end_time, step=0)
-                    for key, value in truncated_metrics_batch.items()
-                ],
-            )
```

Comment:
We can remove lots of complex batching logic from autologging integration implementations because `MlflowAutologgingQueueingClient` handles this in a unified location.